include/crm/common/xml_names.h

```diff
@@ -0,0 +1,455 @@
+/*
+ * Copyright 2004-2024 the Pacemaker project contributors
+ *
+ * The version control history for this file may have further details.
+ *
+ * This source code is licensed under the GNU Lesser General Public License
+ * version 2.1 or later (LGPLv2.1+) WITHOUT ANY WARRANTY.
+ */
+
+#ifndef PCMK__CRM_COMMON_XML_NAMES__H
+#  define PCMK__CRM_COMMON_XML_NAMES__H
+
+#include <crm/common/xml.h>
```

Comment:
I think we should drop this if we're allowed to. The potential problem is that it would break code any external callers that are depending on the transitive include.

I don't see anything in this file that explicitly depends on `xml.h`. However, the `ID()` macro (which may not properly belong here) depends on `nvpair.h`.

Reply:
In fact, if we were to drop this include, then we could include `xml_names.h` in `xml.h` and avoid the literal uses of "master" and "clone" in `crm_map_element_name()`.

Reply:
Should the `xml.h` include within `msg_xml.h` go inside the `PCMK__ALLOW_DEPRECATED` guard or stay outside it?

Reply:
We can actually move the include to msg_xml_compat.h. We could even prevent internal files from including msg_xml.h by adding something like this in the compat block:

    #else
    #error Include xml.h instead of msg_xml.h

By that point we don't even really need a separate compat header, but it makes sense to keep it for consistency.

Reply:
The include is already in `msg_xml_compat.h`, so this would just mean dropping it from `msg_xml.h` and adding the `#error` directive. Works for me.

Reply:
Works for me